### PR TITLE
RbShift modifications to use in tests

### PR DESCRIPTION
**Pull request by mijaros**

### CODE REVIEW

File: lib/rb_shift/service.rb
Comment (jcechace) [inline lib/rb_shift/service.rb]: This is a straight up anti pattern and abuse of unless. 

```ruby
if termination
  # do the thing only when termination is NOT nil
end
```

File: lib/rb_shift/service.rb
Comment (jcechace) [inline lib/rb_shift/service.rb]: Considering this the unless above is even worse.

```ruby
if termination
  # do the thing only when termination is NOT nil
else
  # do the thing only when termination IS nil
end
```

File: lib/rb_shift/project.rb
Comment (jcechace) [inline lib/rb_shift/project.rb:90]: Not sure if possible here, however prefer colon over rocket when constructing hashes

File: lib/rb_shift/client.rb
Comment (jcechace) [inline lib/rb_shift/client.rb]: Nope

**Review by jcechace — CHANGES_REQUESTED** [review]

File: lib/rb_shift/project.rb
Comment (mijaros) [reply lib/rb_shift/project.rb:90]: Well, you can't create symbol by colon from variable, for this you need to use the method
Source is variable which contains string name of source so there is to_sym method used properly.

Comment (jcechace) [inline lib/rb_shift/project.rb]: U was talking about '=>' (rocket) vs ': (colon)' for hashes. The following should work just fine

```ruby
execute 'new-app ', source: path, **opts
```
Considering the following does wok

```ruby
a = 'foo'
{a: 'bar'}

# {:a=>"bar"}
```

File: lib/rb_shift/client.rb
Comment (jcechace) [inline lib/rb_shift/client.rb]: Is this method needed after ```unfold_opts``` was added?

File: lib/rb_shift/client.rb
Comment (jcechace) [inline lib/rb_shift/client.rb]: Personally I would prefer ```find``` instead of ```detect``` but I might be convinced otherwise.

File: lib/rb_shift/client.rb
Comment (jcechace) [inline lib/rb_shift/client.rb:56]: Can't this possibly cause issue by repeatedly making api requests? ```load_projects``` doesn't have any time block.

File: lib/rb_shift/client.rb
Comment (jcechace) [inline lib/rb_shift/client.rb]: Case statement would be more clean here, IMHO.

**Review by jcechace — CHANGES_REQUESTED** [review]

File: lib/rb_shift/client.rb
Comment (mijaros) [reply lib/rb_shift/client.rb:56]: I think that because reloading projects is quite time consuming operation, this cycle will in most cases take one iteration. Only when a cluster will be heavily used the operation will take two cycles.

File: lib/rb_shift/client.rb
Comment (mijaros) [inline lib/rb_shift/client.rb]: @jcechace This comment was marked by rubocop as unnecessary. This commit fixes it, but I'm not sure if this error does not appear on you version of rubocop. If this comment should stay, I will discard this commit, just let me know before merge.

Comment (mijaros) [inline lib/rb_shift/client.rb]: @jcechace This comment was marked by rubocop as unnecessary. This commit fixes it, but I'm not sure if this error does not appear on you version of rubocop. If this comment should stay, I will discard this commit, just let me know before merge.

Comment (jcechace) [reply lib/rb_shift/client.rb]: My version of rubocop unfortunately complains that the ABC complexity of ```get``` method is too high. I will fix the rubocop version in separate pull request.  Lets not mix things  together.

Comment (mijaros) [reply lib/rb_shift/client.rb]: Ok, deleting this commit right away.

File: lib/rb_shift/client.rb
Comment (jcechace) [inline lib/rb_shift/client.rb]:
```ruby 
@token = bearer_token || get_token(url, username, password) 
```     
This would also be idiomatic python btw ;)

File: lib/rb_shift/client.rb
Comment (jcechace) [inline lib/rb_shift/client.rb]: Custom exception would be more appropriate

**Review by jcechace — CHANGES_REQUESTED** [review]

**Comment by jcechace** [comment]
Once the last batch of comments is addressed I think we can merge.  Assigning also to @pestanko to look over this.

**Pull request merged**
